map_claims.go

```diff
@@ -2,150 +2,107 @@ package jwt
 
 import (
 	"encoding/json"
-	"errors"
-	"time"
-	// "fmt"
 )
 
 // MapClaims is a claims type that uses the map[string]interface{} for JSON decoding.
 // This is the default claims type if you don't supply one
 type MapClaims map[string]interface{}
 
-// VerifyAudience Compares the aud claim against cmp.
-// If required is false, this method will return true if the value matches or is unset
-func (m MapClaims) VerifyAudience(cmp string, req bool) bool {
-	var aud []string
-	switch v := m["aud"].(type) {
-	case string:
-		aud = append(aud, v)
-	case []string:
-		aud = v
-	case []interface{}:
-		for _, a := range v {
-			vs, ok := a.(string)
-			if !ok {
-				return false
-			}
-			aud = append(aud, vs)
-		}
-	}
-	return verifyAud(aud, cmp, req)
+// GetExpirationTime implements the Claims interface.
```

Comment:
Above this line could add something like:

```go
var _ Claims = (MapClaims)(nil)
```

Makes it a bit easier to reason that this type is intended to satisfy the `Claims` interface. And then can get rid of the "... implements the Claims interface." comments on the methods.

Reply:
The primary reason was just to have a piece of documentation in the individual functions to understand where they are coming from.